Framework/Geometry/src/Objects/CSGObject.cpp

```diff
@@ -2018,14 +2019,14 @@ V3D CSGObject::generatePointInObject(PseudoRandomNumberGenerator &rng,
       maybePoint = RandomPoint::inGenericShape(
           *this, rng, maxAttempts - bruteForceAttempts);
       if (!maybePoint) {
-        throw std::runtime_error("Unable to generate point in object after " +
-                                 std::to_string(maxAttempts) + " attempts");
+        return false;
+      } else {
```

Comment:
This is more of a style thing, but I find it easier to read
```
if (!maybePoint){
    return false;
 }  // Else block not here since return means we can't go below
  point = *maybePoint;
  break;
```

Reply:
The change to use boost::optional means this if statement has been removed now so don't think there's a change required any more